pkg/resolver/testHelper_test.go

```diff
@@ -131,35 +162,97 @@ func stringInSlice(a string, list []string) bool {
 	return false
 }
 
-// Load mock data based on input filters
+// Only load mock data items if the input filters conditions are satisfied
 func useInputFilterToLoadData(mockDataFile string, input *model.SearchInput, item interface{}) bool {
-	if !strings.Contains(mockDataFile, "rel") { // To load resources file
-		for _, filter := range input.Filters {
-			if len(filter.Values) > 0 {
-				values := pointerToStringArray(filter.Values)
-				uid := item.(map[string]interface{})["uid"]
+	// var destkind string
+	var relatedValues []string
+
+	if len(input.RelatedKinds) > 0 { //If relatedKinds filter is present
+		relatedValues = pointerToStringArray(input.RelatedKinds)
+		// data := item.(map[string]interface{})["properties"].(map[string]interface{})
+		destkind := item.(map[string]interface{})["DestKind"].(string)
+		// destkind = data["DestKind"].(string)
+		if stringInSlice(destkind, relatedValues) {
+			return true
+		} else {
+			return false // If the resource kind is not in RelatedKinds, do not load it
+		}
+
+	}
+
+	for _, filter := range input.Filters {
+		if len(filter.Values) > 0 {
+			values := pointerToStringArray(filter.Values) //get the filter values
+			op, values := getOperator(values)             //get the operator if filter property åis a number
+			if op == "" {
+				_, values = getDateFilter(values) // get the filter values if property is a date
+			}
+			uid := item.(map[string]interface{})["uid"]
+
+			if filter.Property == "cluster" {
 				cluster := strings.Split(uid.(string), "/")[0]
-				if filter.Property == "cluster" {
-					if !stringInSlice(cluster, values) {
-						return false // If the filter value is not in resource, do not load it
+				if !stringInSlice(cluster, values) {
+					return false // If the filter value is not in resource, do not load it
+				}
+			} else {
+				data := item.(map[string]interface{})["properties"].(map[string]interface{})
+
+				if data[filter.Property] == nil { // if required property is not set, don't load the item
+					return false
+				}
+				var filterValue int
+				var err error
+				if op != "" {
+					filterValue, err = strconv.Atoi(values[0]) // if the property is a number, get the first value
+					if err != nil {
+						fmt.Println("Error converting value to int", err)
 					}
-				} else {
-					data := item.(map[string]interface{})["properties"].(map[string]interface{})
+				}
 
-					if !stringInSlice(data[filter.Property].(string), values) {
+				switch op {
+				case "<":
+					if int(data[filter.Property].(float64)) < filterValue {
+						return true
+					}
+					return false
```

Comment:
This is the same and more compact.

```suggestion
					return int(data[filter.Property].(float64)) < filterValue
```